google-cloud-firestore/lib/google/cloud/firestore/bulk_writer_exception.rb

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # #  BulkWriterException
+      #
+      # A BulkWriterException object refers to the error that will be thrown
+      # incase a BulkWriterOperation fails in all the attempts.
+      #
+      class BulkWriterException < StandardError
+        attr_reader :status
+        attr_reader :message
+        attr_reader :details
+
+        def initialize status
+          @status = status.code
```

Comment:
Attribute name wasn't set as `status_code` to prevent confusing it directly with the HTTP status codes, becasue `status.code` is an integer b/w `0-15`. These codes do have a direct HTTP mapping which can be found [here](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). 
Open to the updating the attribute name, but ig adding these details in the documentation will suffice.